subprojects/docs/src/docs/userguide/migration/upgrading_version_6.adoc

```diff
@@ -219,6 +219,13 @@ This may cause test failures when the test expects particular file paths.
 
 If the test uses `GradleRunner.withTestKitDir(...)`, it is unaffected.
 
+==== File system watching with TestKit on Windows
+
+The file system watching implementation on Windows adds a lock to the root project directory in order to watch for changes in there.
+This may cause problems in functional tests when the root project directory is within a temporary directory which is deleted after the test.
```

Comment:
I don't have a good suggestion, but this is a very specific description. I wonder if we could provide a more general one that is also easier to follow somehow?

Maybe keeping the phrasing would work if we reframed this as just one example case where the problem might occur?

Reply:
Also, I wonder if specifying that this is about _functional_ tests matters. TestKit is for running builds as part of tests. What you call those kinds of tests isn't very important here, and I don't think we should be prescriptive.

Reply:
I updated the upgrading doc.